src/java.base/unix/classes/sun/nio/fs/UnixFileStore.java

```diff
@@ -179,6 +179,11 @@ public Object getAttribute(String attribute) throws IOException {
      * @return <code>true</code> if enabled, <code>false</code> if disabled or unable to determine
      */
     protected boolean isExtendedAttributesEnabled(UnixPath path) {
+        if (!UnixNativeDispatcher.xattrSupported()) {
+            // avoid I/O if native code doesn't support xattr
+            return false;
+        }
+
```

Comment:
Add SUPPORTS_XATTR is a good idea.
Style-wise, it might be a bit neater to use if (UnixNativeDispatcher.xattrSupported()) { ...} so there will only one "return false" code path.
Looking at this code now make me wonder about the close(fd) for the case that the open fails. It should be checking if fd or there should be a nested try-finally so that close is only invoked when the open succeeds.

Reply:
> Style-wise, it might be a bit neater to use if (UnixNativeDispatcher.xattrSupported()) { ...} so there will only one "return false" code path.

I prefer not to avoid nested blocks for improved readability, but thats mostly a matter of taste. If you like, I can reverse the "if".

> Looking at this code now make me wonder about the close(fd) for the case that the open fails. It should be checking if fd or there should be a nested try-finally so that close is only invoked when the open succeeds.

True. But I guess that would be off-topic in this PR?

Reply:
It's somewhat subjective. What you have is fine, I just would have done is differently to avoid too many return paths.

I think we should clean up the close(-1) case while we are in the area, separate PR of course.

Reply:
All right, will you create another bug report for the `close(-1)` issue?

Reply:
I just wanted to create another issue, but then I noticed that `close(-1)` is allowed as per spec: https://github.com/openjdk/jdk/blob/45e1bab87ccebd498d4f927a008dd89a47dd177b/src/java.base/unix/classes/sun/nio/fs/UnixNativeDispatcher.java#L90-L97

We can still change this, of course, but I wouldn't consider this a necessary change for this PR and would therefore suggest to deal with this another time.

Reply:
Okay, I had forgotten that that UnixNativeDispatcher.close(-1) is a no-op.

Reply:
I guess we're good to merge, then?